routes/problems_id_solutions.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const datastore             = require('../modules/datastore.js');
+const auth                  = require('../modules/auth.js');
+const accessControl         = require('../modules/access_control.js');
+
+const SOLUTION_KIND         = 'Solution';
+const DEFAULT_LIMIT         = 15;
+const DEFAULT_SORT          = 'complexity';
+const DEFAULT_IS_DESCENDING = false;
+
+
+/**
+ * Utility function to get a list of submitted solutions to display.
+ * Note that the code is not included in this query as it will be
+ * handled by POST request handler, which will check if the client
+ * has enough privilege to view the code.
+ **/
+async function getSolutions(problemId, limit, sortBy, isDescending) {
+  const query = datastore
+    .createQuery(SOLUTION_KIND)
+    .filter('problemId', '=', parseInt(problemId))
+    .order(sortBy, {
+      descending: isDescending,
+    })
+    .limit(parseInt(limit))
+    .select(['email', 'complexity']);
+
+  return (await datastore.runQuery(query))[0];
+}
+
+async function getSubmittedCode(problemId, email) {
```

Comment:
for non-mvp, getSolutions and getSubmittedCode or their callers need to handle errors.

Reply:
Noted, thanks!
I'll submit this issue in our Buganizer.